trie/validation.py

```diff
@@ -38,3 +42,10 @@ def validate_is_node(node):
                 validate_length(sub_node, 32)
     else:
         raise ValidationError("Invalid Node: {0}".format(node))
+
+
+def validate_is_bin_node(node):
+    if node == BLANK_HASH or node[0] == BRANCH_TYPE or node[0] == KV_TYPE or node[0] == LEAF_TYPE:
```

Comment:
if `node[0] in {BRANCH_TYPE, KV_TYPE, LEAF_TYPE}` is a bit more concise way to express this logic.